docs/rfc/rfc-1.adoc

```diff
@@ -435,13 +435,49 @@ As time passes and operators drop out of the system, a wallet becomes at risk
 of being able to meet the 51-of-100 threshold to produce signatures.
 Additionally, we want to avoid situations where operators are the custodians of
 a wallet for extended periods. To avoid these issues, once a wallet is older
-than the `wallet_max_age`, or if it drops below the liveness threshold (say, below 70 on
-a <<heartbeat,heartbeat>>), we motion to transfer the funds to another randomly
-selected wallet.
+than the `wallet_max_age`, or if it drops below the liveness threshold (say,
+below 70 on a <<heartbeat,heartbeat>>), we motion to
+<<closing-a-wallet,transfer the funds>> to another randomly selected wallet.
 
 Once a wallet no longer has funds and is not the primary wallet for new
-deposits, it can be closed and operators are no longer required to maintain
-it.
+deposits, it can be <<closing-a-wallet,closed>> and operators are no longer
+required to maintain it.
+
+[[closing-a-wallet]]
+==== Closing A Wallet
+Governable Parameters:
+
+- `wallet_min_btc`: The smallest amount of BTC a wallet can hold before we
+  attempt to close the wallet and transfer the funds to a randomly selected
+  wallet.
+
+When a wallet fails a <<heartbeat,heartbeat>> or ends a
+<<redemption,redemption>> with less than `wallet_min_btc` remaining, the
+operators begin the process of closing it. If there are still funds remaining,
+the <<operator-only,first operator>> posts a <<operator-only,reimbursable>>
+transaction to the ethereum chain declaring an intention to close the wallet.
+
+That operator proposes a BTC fee (as in <<sweeping,sweeping>>) and an ethereum
+block height commitment, and they use that to construct a P2PKH transaction
+moving the wallet's main UTXO to another randomishly chosen wallet where the
+source of randomness comes from the ethereum block hash at the pre-commited
+block height.
```

Comment:
I do not understand the pre-committed block height, can you please elaborate?

What I think we could do instead of an additional commitment is that when someone (does not even have to be an operator from that wallet - this is *public-knowledge*) notifies that a wallet has funds below the dust threshold, they submit a transaction to the chain to select another wallet where the funds should go to.

We take the blockhash (I think this is sufficient) and % N where N is the number of most active wallets we are interested in. We record blockhash % N as the target wallet assuming the amount hold by that wallet is above the dust threshold or that wallet is active; if not we decrement to move to the more recent wallets.

For example:

wallet 20 [ active ] 0.01 BTC
wallet 19 [ active ] 200 BTC
wallet 18 [ inactive ] 198 BTC
wallet 17 [inactive ] 194 BTC
...
wallet 3 [ inactive ] 10 BTC
wallet 2 [ inactive ] 5 BTC
wallet 1 [ inactive ] 0.01 BTC

With N = 4, we would be selecting from wallets 20-17 and all wallets qualify; they are either above the dust threshold (17-19) or they are below the dust threshold but active (20).

Reply:
I like this scheme - so the public just *points out* after a redemption that the funds need to be transferred, or the operators *point out* after a failed heartbeat that the funds need to be transferred. No one but the miners are able to change the hash of this "hey we need to transfer" declaration, and it carries two benefits:

a) it determines the target wallet on-chain
b) it sets the beginning of the time-out on-chain

:+1:

Reply:
addressed in [e4717ca](https://github.com/keep-network/tbtc-v2/pull/127/commits/e4717caa5fa2862dce4f2e184f670b441c6fc43b)